addons/payment_alipay/models/payment_transaction.py

```diff
@@ -0,0 +1,147 @@
+                         f"state: {self.state}, received data: {data}")
+            return
+        if float_compare(float(data.get('total_fee', '0.0')), (self.amount + self.fees), 2) != 0:
+            # mc_gross is amount + fees
+            _logger.error(f"Alipay: The paid amount ({data.get('total_fee', '0.0')}) does not match the "
+                          f"total + fees ({self.amount} + {self.fees}) for ref {self.reference}")
+            raise ValidationError("Alipay: " + _("The paid amount does not match the total + fees."))
+        if self.acquirer_id.alipay_payment_method == 'standard_checkout':
+            if data.get('currency') != self.currency_id.name:
+                _logger.error("Alipay: The currency returned by Alipay %(rc)s does not match the "
+                              "transaction currency %(tc)s.)",
+                              rc=data.get('currency'), tc=self.currency_id.name)
+                raise ValidationError("Alipay: " + _("The currency returned by Alipay %(rc)s does not match the "
+                                                     "transaction currency %(tc)s.",
+                                                     rc=data.get('currency'), tc=self.currency_id.name))
+        elif data.get('seller_email') != self.acquirer_id.alipay_seller_email:
+            _logger.error(f"Alipay: The seller email ({data.get('seller_email')}) does not match the configured "
+                          f"Alipay account ({self.acquirer_id.alipay_seller_email}).")
+            raise ValidationError("Alipay: " + _("The seller email does not match the configured Alipay account."))
+
+        status = data.get('trade_status')
+        if status in ['TRADE_FINISHED', 'TRADE_SUCCESS']:
+            self._set_done()
+        elif status == 'TRADE_CLOSED':
+            _logger.info('Received transaction data for Alipay payment %s: set as Canceled' % self.reference)
+            self._set_canceled()
+        else:
+            _logger.info(f'Received unrecognized status for Alipay payment {self.reference}: {status}, set as error')
+            self._set_error("Alipay: " + _(
+                "received data with invalid transaction status: %(tx_status)s", tx_status=status
+            ))
+
+    def _get_specific_rendering_values(self, processing_values):
+        if self.acquirer_id.provider != 'alipay':
+            return super()._get_specific_processing_values(processing_values)
+        base_url = self.acquirer_id._get_base_url()
+        alipay_tx_values = {
+            '_input_charset': 'utf-8',
+            'notify_url': urls.url_join(base_url, AlipayController._notify_url),
+            'out_trade_no': processing_values.get('reference'),
+            'partner': self.acquirer_id.alipay_merchant_partner_id,
+            'return_url': urls.url_join(base_url, AlipayController._return_url),
+            'subject': processing_values.get('reference'),
+            'total_fee': processing_values.get('amount') + self.fees
```

Comment:
```suggestion
            'out_trade_no': processing_values['reference'],
            'partner': self.acquirer_id.alipay_merchant_partner_id,
            'return_url': urls.url_join(base_url, AlipayController._return_url),
            'subject': processing_values.get['reference'],
            'total_fee': processing_values.get['amount'] + self.fees
```
This shows that the key is always expected in the dict and ensures that we will never have `None + self.fees` imo